inc/ctest.h

```diff
@@ -204,55 +204,68 @@ typedef void* void_ptr;
-extern C_LINKAGE int float_Compare(float left, float right);
-extern C_LINKAGE int double_Compare(double left, double right);
-extern C_LINKAGE int long_double_Compare(long double left, long double right);
-extern C_LINKAGE int char_ptr_Compare(const char* left, const char* right);
-extern C_LINKAGE int wchar_ptr_Compare(const wchar_t* left, const wchar_t* right);
-extern C_LINKAGE int void_ptr_Compare(const void * left, const void* right);
-extern C_LINKAGE int unsigned_long_Compare(unsigned long left, unsigned long right);
-
-#if defined CTEST_USE_STDINT
-extern C_LINKAGE void uint8_t_ToString(char* string, size_t bufferSize, uint8_t val);
-extern C_LINKAGE void int8_t_ToString(char* string, size_t bufferSize, int8_t val);
-extern C_LINKAGE void uint16_t_ToString(char* string, size_t bufferSize, uint16_t val);
-extern C_LINKAGE void int16_t_ToString(char* string, size_t bufferSize, int16_t val);
-extern C_LINKAGE void uint32_t_ToString(char* string, size_t bufferSize, uint32_t val);
-extern C_LINKAGE void int32_t_ToString(char* string, size_t bufferSize, int32_t val);
-extern C_LINKAGE void uint64_t_ToString(char* string, size_t bufferSize, uint64_t val);
-extern C_LINKAGE void int64_t_ToString(char* string, size_t bufferSize, int64_t val);
-extern C_LINKAGE int uint8_t_Compare(uint8_t left, uint8_t right);
-extern C_LINKAGE int int8_t_Compare(int8_t left, int8_t right);
-extern C_LINKAGE int uint16_t_Compare(uint16_t left, uint16_t right);
-extern C_LINKAGE int int16_t_Compare(int16_t left, int16_t right);
-extern C_LINKAGE int uint32_t_Compare(uint32_t left, uint32_t right);
-extern C_LINKAGE int int32_t_Compare(int32_t left, int32_t right);
-extern C_LINKAGE int uint64_t_Compare(uint64_t left, uint64_t right);
-extern C_LINKAGE int int64_t_Compare(int64_t left, int64_t right);
-#endif
 
 extern C_LINKAGE char* ctest_sprintf_char(const char* format, ...);
 extern C_LINKAGE void ctest_sprintf_free(char* string);
 
+#define CTEST_DECLARE_EQUALITY_ASSERTION_FUNCTIONS_FOR_TYPE(type) \
+extern C_LINKAGE void MU_C2(type,_AssertAreEqual)(type left, type right, char* ctest_message); \
+extern C_LINKAGE void MU_C2(type,_AssertAreNotEqual)(type left, type right, char* ctest_message);
+
+#define CTEST_ASSERT_ARE_EQUAL_IMPL_FOR_TYPE(type) \
+    char expectedString[1024]; \
+    char actualString[1024]; \
+    MU_C2(type,_ToString)(expectedString, sizeof(expectedString), left); \
+    MU_C2(type,_ToString)(actualString, sizeof(actualString), right); \
+    if (MU_C2(type,_Compare)(left, right)) \
+    { \
+        LogError("  Assert failed in line %d %s Expected: %s, Actual: %s\n", __LINE__, (ctest_message == NULL) ? "" : ctest_message, expectedString, actualString); \
+        ctest_sprintf_free(ctest_message); \
+        if (g_CurrentTestFunction != NULL) *g_CurrentTestFunction->TestResult = TEST_FAILED; \
+        do_jump(&g_ExceptionJump, expectedString, actualString); \
+    } \
+    ctest_sprintf_free(ctest_message);
```

Comment:
Was considering whether it would make sense to have a common macro here and have the common macro have a bool argument that could be used like:

#define CTEST_ASSERT_ARE_EQUAL_IMPL_FOR_TYPE(type, check_for_is_equal) \
...
if (MU_C2(type,_Compare)(left, right) == check_for_is_equal) \

That was the code would not be duplicated.